wolfcrypt/src/evp.c

```diff
@@ -6941,6 +6953,39 @@ int wolfSSL_EVP_MD_size(const WOLFSSL_EVP_MD* type)
     return BAD_FUNC_ARG;
 }
 
+int wolfSSL_EVP_MD_pkey_type(const WOLFSSL_EVP_MD* type)
+{
+    int ret = BAD_FUNC_ARG;
+
+    WOLFSSL_ENTER("wolfSSL_EVP_MD_pkey_type");
+
+    if (type != NULL) {
+        if (XSTRNCMP(type, "MD5", 3) == 0) {
```

Comment:
Any reason not to use a lookup via the `md_tbl` table with for loop?

Reply:
`md_tbl` has different NIDs. For example, it has `NID_md5`, which is 4, whereas this returns `NID_md5WithRSAEncryption` for MD5, which is 8.